### ISSUE TITLE
How do I set a default edge

### ISSUE THREAD

**Issue by camer314** [wbkd/react-flow#1364]
This seems like such a simple thing but I cannot figure out how to set the edge that is used by default when the user creates a connection.

Basically I want my default edge to have an arrowhead without having to write a custom one

**Comment by moklick** [comment]
Are you speaking about the connection line or an edge? For an edge you can pass the option [`arrowHeadType`](https://reactflow.dev/docs/api/edges/) if you need a connection line with an arrow you would need to create a custom connection line: https://reactflow.dev/examples/custom-connectionline/

**Comment by camer314** [comment]
Yes sorry. Well maybe I dont appreciate the difference between an edge and a connector. For example, an input node clearly only has an output connector so when its connected to another node I would like to see the arrowhead going in that direction.

But then if i delete that connector and then create a new one now leaving that node into another node then I want the arrow to show the direction of that connector.

In other words, is it possible to draw an arrow always in the direction you connect nodes to indicate direction of flow?

**Comment by camer314** [comment]
Further update to this. I think I didnt understand the way these edges work. I thought both the top and bottom edge can be either an input or output, seems like the input is always the top and output always the bottom, which makes sense.

I also figured out I could implement the onConnect callback and override the default edge there:

```
const onConnect = (params: Connection | Edge) => setElements((els) => {
    var edge: Edge = {
      id: getId(),
      source: params.source ?? '',
      target: params.target ?? '',
      arrowHeadType: ArrowHeadType.ArrowClosed
    };
    return addEdge(edge, els);
  });
```

**Comment by moklick** [comment]
1. "seems like the input is always the top and output always the bottom" this is true for the builtin `default` node type. You can add custom nodes with sources and targets on every side of the node.
2. Yes. This is how you overwrite it. You could simplify it by using the passed params:
```js
const onConnect = (params: Connection | Edge) => setElements((els) => {
    const edge: Edge = {
     ...params,
      arrowHeadType: ArrowHeadType.ArrowClosed
    };
    return addEdge(edge, els);
  });
```
